python/ray/tune/examples/tf_distributed_keras_example.py

```diff
@@ -0,0 +1,106 @@
+"""
+Adapted from
+https://www.tensorflow.org/tutorials/distribute/multi_worker_with_keras
+"""
+import json
+import os
+import argparse
+import tensorflow as tf
+import numpy as np
+from ray import tune
+from ray.tune.schedulers import AsyncHyperBandScheduler
+from ray.tune.integration.keras import TuneReportCheckpointCallback
+from ray.tune.integration.tensorflow import DistributedTrainableCreator
+
+
+def mnist_dataset(batch_size):
+    (x_train, y_train), _ = tf.keras.datasets.mnist.load_data()
+    # The `x` arrays are in uint8 and have values in the range [0, 255].
+    # You need to convert them to float32 with values in the range [0, 1]
+    x_train = x_train / np.float32(255)
+    y_train = y_train.astype(np.int64)
+    train_dataset = tf.data.Dataset.from_tensor_slices(
+        (x_train, y_train)).shuffle(60000).repeat().batch(batch_size)
+    return train_dataset
+
+
+def build_and_compile_cnn_model(config):
+    model = tf.keras.Sequential([
+        tf.keras.Input(shape=(28, 28)),
+        tf.keras.layers.Reshape(target_shape=(28, 28, 1)),
+        tf.keras.layers.Conv2D(32, 3, activation="relu"),
+        tf.keras.layers.Flatten(),
+        tf.keras.layers.Dense(config.get("hidden", 128), activation="relu"),
+        tf.keras.layers.Dense(10)
+    ])
+    model.compile(
+        loss=tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True),
+        optimizer=tf.keras.optimizers.SGD(learning_rate=config.get("lr", 0.05),
+                                          momentum=config.get("momentum",
+                                                              0.5)),
+        metrics=["accuracy"])
+    return model
+
+
+def train_mnist(config, checkpoint_dir=None):
+    strategy = tf.distribute.experimental.MultiWorkerMirroredStrategy()
+    per_worker_batch_size = 64
+    tf_config = json.loads(os.environ["TF_CONFIG"])
+    num_workers = len(tf_config["cluster"]["worker"])
```

Comment:
hmm, can we avoid this? maybe have a:

```
ray.tune.integration.tensorflow.get_num_workers()
```
?

Reply:
Updated.. this is more like a common util function for all distributed tf training tho.

Reply:
yeah, i guess you could do integration.tensorflow.util.get_num_workers